pkg/cloudprovider/gce/gce.go

```diff
@@ -379,11 +383,53 @@ func (gce *GCECloud) CreateTCPLoadBalancer(name, region string, externalIP net.I
 		return nil, err
 	}
 
+	allowedPorts := make([]string, len(ports))
+	for ix := range ports {
+		allowedPorts[ix] = strconv.Itoa(ports[ix].Port)
+	}
+
+	hostTag := gce.computeHostTag(hosts[0])
+
+	firewall := &compute.Firewall{
+		Name:         makeFirewallName(name),
+		Description:  fmt.Sprintf("KubernetesAutoGenerated_OnlyAllowTrafficForDestinationIP_%s", fwd.IPAddress),
+		Network:      gce.gceNetworkName(),
+		SourceRanges: []string{"0.0.0.0/0"},
+		TargetTags:   []string{hostTag},
+		Allowed: []*compute.FirewallAllowed{
+			{
+				IPProtocol: "tcp",
+				Ports:      allowedPorts,
+			},
+		},
+	}
+	if op, err = gce.service.Firewalls.Insert(gce.projectID, firewall).Do(); err != nil {
+		return nil, err
+	}
+	if op.Status == "DONE" && op.Error != nil {
+		// An error occurred
+		if op.HttpErrorStatusCode == http.StatusConflict {
+			glog.V(4).Infof("Detected an already existing firewall: %s", firewall.Name)
```

Comment:
Good point.  OK.  I'm a little nervous about it still, but it just needs testing and shake-out